### ISSUE TITLE
error: #include nested too deeply

### ISSUE THREAD

**Issue by janaranjanahl** [ThrowTheSwitch/Ceedling#677]
I get following error when I include the path to the include

ERROR
```
                 from /opt/fsl-imx-fb/5.10-hardknott/sysroots/cortexa7t2hf-neon-poky-linux-gnueabi/usr/include/sys/unistd.h:1,
                 from /opt/fsl-imx-fb/5.10-hardknott/sysroots/cortexa7t2hf-neon-poky-linux-gnueabi/usr/include/sys/unistd.h:1,
                 from /opt/fsl-imx-fb/5.10-hardknott/sysroots/cortexa7t2hf-neon-poky-linux-gnueabi/usr/include/sys/unistd.h:1,
                 from /opt/fsl-imx-fb/5.10-hardknott/sysroots/cortexa7t2hf-neon-poky-linux-gnueabi/usr/include/sys/unistd.h:1,
```

**to keep the post short I removed 100 > lines :(**

```
                 from /opt/fsl-imx-fb/5.10-hardknott/sysroots/cortexa7t2hf-neon-poky-linux-gnueabi/usr/include/sys/unistd.h:1,
                 from /opt/fsl-imx-fb/5.10-hardknott/sysroots/cortexa7t2hf-neon-poky-linux-gnueabi/usr/include/sys/unistd.h:1,
                 from /opt/fsl-imx-fb/5.10-hardknott/sysroots/cortexa7t2hf-neon-poky-linux-gnueabi/usr/include/sys/unistd.h:1,
                 from include/comms/evcc_can.h:12,
                 from build/temp/_test_can.c:5:
/opt/fsl-imx-fb/5.10-hardknott/sysroots/cortexa7t2hf-neon-poky-linux-gnueabi/usr/include/sys/unistd.h:1:20: error: #include nested too deeply
 #include <unistd.h>
                    ^
ERROR: Shell command failed.
> Shell executed command
```

YML file
---
:project:
  :use_exceptions: FALSE
  :use_test_preprocessor: TRUE
  :use_auxiliary_dependencies: TRUE
  :build_root: build
  :test_file_prefix: test_
  :which_ceedling: gem
  :ceedling_version: 0.31.1
  :default_tasks:
    - test:all
:environment:
:extension:
  :executable: .out
:paths:
  :test:
    - +:test/**
  :source:
    - src/**
  :include:
    - include/**
    - /opt/fsl-imx-fb/5.10-hardknott/sysroots/cortexa7t2hf-neon-poky-linux-gnueabi/usr/include/unistd.h
    - /opt/fsl-imx-fb/5.10-hardknott/sysroots/cortexa7t2hf-neon-poky-linux-gnueabi/usr/include/sys/**
  :support:
  :libraries: []
:files:
  :include:
    - +:/opt/fsl-imx-fb/5.10-hardknott/sysroots/cortexa7t2hf-neon-poky-linux-gnueabi/usr/include/include/**
:defines:

It is interesting to note that unistd.h include itself in  the file
![image](https://user-images.githubusercontent.com/23205104/166062303-cd6e4787-1913-4e68-ba1b-a0ebb2b09722.png)

is there a way to avoid nested too deeply issue?

**Comment by mvandervoord** [comment]
I don't believe there is anything Ceedling can do about this issue. Your compiler is issuing the error that your library has an infinite recursive include loop without any type of guard. Sometimes compilers have short-circuits for this... apparently this one does not? The best solution would be to fix the bug in the library, but I suspect this is a "standard library" and that probably isn't simple for you.

**Comment by janaranjanahl** [comment]
That is unfortunate. Thanks for prompt reply!

**Comment by janaranjanahl** [comment]
I will close the issue at this time

**Comment by mvandervoord** [comment]
Out of curiosity, are you using some form of gcc? How does this not happen when you're compiling a release version of your code? You might want to look at the flags being passed to your compiler during release... maybe one of them will give you a clue as to how to avoid this?

**Comment by janaranjanahl** [comment]
My application is cross compiled with `arm-poky-linux-gnueabi-gcc` .

But we run Ceedling with default gcc to run with Ubuntu 

this is our tools section
```
:tools_test_linker:
  :arguments:
    - -lpthread
    - -lsocketcan
:tools_gcov_linker:
  :arguments:
    - -lpthread
    - -lsocketcan
```
